### PR TITLE
feat!: apply retroactive deprecations

### PR DESCRIPTION
**Pull request by agoose77**
- Raise error for `parameters` that are neither `None` or `{}` in `EmptyForm` et al.
  - We still allow `parameters=None` and `parameters={}` in the constructor arguments, to ensure that content-naive workflows still work.
- Raise error for conversions of unions to backend arrays

### CODE REVIEW

File: src/awkward/contents/emptyarray.py [hunk 1]
```diff
@@ -85,29 +82,26 @@ def copy(
...
     def __copy__(self):
         return self.copy()
 
     def __deepcopy__(self, memo):
-        return self.copy(parameters=copy.deepcopy(self._parameters, memo))
+        return self.copy()
 
     @classmethod
     def simplified(cls, *, parameters=None, backend=None):
-        if not (parameters is None or len(parameters) == 0):
```
Comment (agoose77) [inline src/awkward/contents/emptyarray.py:105]: This is a more permissive test for the deprecations, but I think it's okay to break empty dicts here too.

**Review by jpivarski — APPROVED** [review]
Right: this makes good on the deprecation promise of forbidding `EmptyArray` with `parameters`.
